core/src/main/java/org/elasticsearch/cluster/service/InternalClusterService.java

```diff
@@ -130,6 +133,9 @@ public InternalClusterService(Settings settings, DiscoveryService discoveryServi
         localNodeMasterListeners = new LocalNodeMasterListeners(threadPool);
 
         initialBlocks = ClusterBlocks.builder().addGlobalBlock(discoveryService.getNoMasterBlock());
+
+        taskManager = new TaskManager(settings);
+        transportService.setTaskManager(taskManager);
```

Comment:
oh I see cool, now that we know who has to be initialized first can we move the construction into `TransportService` and then just call the getter here so we have all references final and non-volatile?